standards_lab/api/views.py

```diff
@@ -140,15 +148,27 @@ def delete(self, *args, **kwargs):
 
 class ProjectDownloadFile(View):
     def get(self, request, *args, **kwargs):
+        type = kwargs["type"]
         project = get_project_config(kwargs["name"])
 
-        # Downloading is limited to files tracked in our project
-        if not kwargs["file_name"] in project.get("schemaFiles", []) + project.get(
-            "dataFiles", []
-        ):
+        # check only allowed types & files tracked in our project
+        if type == "data":
+            if not kwargs["file_name"] in project.get("dataFiles", []):
+                raise Http404
+        elif type == "schema":
+            if not kwargs["file_name"] in project.get("schemaFiles", []):
+                raise Http404
+        else:
             raise Http404
 
-        file_path = os.path.join(project["path"], kwargs["file_name"])
+        file_path = os.path.join(
+            project["path"],
+            ("datas" if type == "data" else "schemas"),
```

Comment:
Same as previous comment on use of "else"  here

Reply:
In this case the if/elif/else above has already checked that a correct type has been passed, and thrown a 404 if not.